StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/PropertySummaryDocumentationAnalyzer.cs

```diff
@@ -225,51 +228,85 @@ private static void AnalyzeSummaryElement(SyntaxNodeAnalysisContext context, Xml
-                    else if (!text.StartsWith(startingTextGets, StringComparison.OrdinalIgnoreCase))
-                    {
-                        diagnosticProperties.Add(ExpectedTextKey, startingTextGets);
-                        context.ReportDiagnostic(Diagnostic.Create(SA1623Descriptor, diagnosticLocation, diagnosticProperties.ToImmutable(), startingTextGets));
+                        ReportSa1623(context, diagnosticLocation, diagnosticProperties, text, expectedStartingText: startingTextGets, unexpectedStartingText1: startingTextSets, unexpectedStartingText2: startingTextGetsOrSets);
                     }
                 }
             }
             else if (setterVisible)
             {
-                if (startsWithGetOrSet)
+                if (getter != null)
                 {
-                    diagnosticProperties.Add(ExpectedTextKey, startingTextSets);
-                    diagnosticProperties.Add(TextToRemoveKey, startingTextGetsOrSets);
-                    context.ReportDiagnostic(Diagnostic.Create(SA1624Descriptor, diagnosticLocation, diagnosticProperties.ToImmutable(), "set", startingTextSets));
+                    // Both getter and setter exist, but only setter is visible.
+                    if (!prefixIsSets)
+                    {
+                        if (prefixIsGetsOrSets)
+                        {
+                            ReportSa1624(context, diagnosticLocation, diagnosticProperties, accessor: "set", expectedStartingText: startingTextSets, startingTextToRemove: startingTextGetsOrSets);
+                        }
+                        else
+                        {
+                            ReportSa1623(context, diagnosticLocation, diagnosticProperties, text, expectedStartingText: startingTextSets, unexpectedStartingText1: startingTextGets);
+                        }
+                    }
                 }
-                else if (!text.StartsWith(startingTextSets, StringComparison.OrdinalIgnoreCase))
+                else
                 {
-                    diagnosticProperties.Add(ExpectedTextKey, startingTextSets);
-                    context.ReportDiagnostic(Diagnostic.Create(SA1623Descriptor, diagnosticLocation, diagnosticProperties.ToImmutable(), startingTextSets));
+                    // Setter exists and is visible. Getter does not exist.
+                    if (!prefixIsSets)
+                    {
+                        ReportSa1623(context, diagnosticLocation, diagnosticProperties, text, expectedStartingText: startingTextSets, unexpectedStartingText1: startingTextGetsOrSets, unexpectedStartingText2: startingTextGets);
+                    }
                 }
             }
         }
+
+        private static void ReportSa1623(SyntaxNodeAnalysisContext context, Location diagnosticLocation, ImmutableDictionary<string, string>.Builder diagnosticProperties, string text, string expectedStartingText, string unexpectedStartingText1, string unexpectedStartingText2 = null)
```

Comment:
I would prefer `ReportSA1623` here, to more clearly preserve the link with the identifier that is reported.